app/controllers/display_controller.rb

```diff
@@ -75,6 +75,48 @@ def read_all_works
     session[:col_id] = @collection.slug
   end
 
+  def display_page
+    # Set meta information for web crawlers and archival
+    @page_title = "#{@page.title || "Page #{@page.position}"} - #{@work.title} - #{@collection.title}"
+    @meta_description = "Transcript of #{@page.title || "page #{@page.position}"} from #{@work.title} in the #{@collection.title} collection."
+    @meta_keywords = [@work.title, @collection.title, @page.title, "transcript", "transcription", "historical document"].compact.join(", ")
+    
+    # Generate structured data for better content understanding
+    @structured_data = {
+      "@context" => "https://schema.org",
+      "@type" => "DigitalDocument",
+      "name" => @page.title || "Page #{@page.position}",
+      "description" => @meta_description,
+      "text" => @page.source_text,
+      "inLanguage" => @collection.text_language || "en",
+      "isPartOf" => {
+        "@type" => "Book",
+        "name" => @work.title,
+        "author" => @work.author,
+        "dateCreated" => @work.document_date,
+        "isPartOf" => {
+          "@type" => "Collection",
+          "name" => @collection.title,
+          "description" => @collection.intro_block
+        }
+      },
+      "url" => request.original_url,
+      "dateModified" => @page.updated_at&.iso8601,
+      "publisher" => {
+        "@type" => "Organization", 
+        "name" => "FromThePage"
```

Comment:
The publishing organization should be the collection owner's `display_name`, not FromThePage.